flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkOperatorConfiguration.java

```diff
@@ -26,26 +26,33 @@
 @Value
 public class FlinkOperatorConfiguration {
 
-    int reconcileIntervalInSec;
-
-    int portCheckIntervalInSec;
-
-    int savepointTriggerGracePeriodInSec;
+    int reconcileIntervalSeconds;
+    int progressCheckIntervalSeconds;
+    int restApiReadyDelaySeconds;
+    int savepointTriggerGracePeriodSeconds;
 
     public static FlinkOperatorConfiguration fromConfiguration(Configuration operatorConfig) {
-        int reconcileIntervalInSec =
+        int reconcileIntervalSeconds =
                 operatorConfig.getInteger(
                         OperatorConfigOptions.OPERATOR_RECONCILER_RESCHEDULE_INTERVAL_IN_SEC);
-        int portCheckIntervalInSec =
+
+        int restApiReadyDelaySeconds =
```

Comment:
Out the scope of this PR.

To be honest, I do not like the `restApiReadyDelaySeconds` configuration. This make the `JobManagerDeploymentStatus.READY` not really ready for accepting REST API calls. This will be problem if we are running session cluster.

Reply:
Why would this affect session cluster? There is a delay between the port becoming available and the actual rest server being ready which absent any other reliable technique to check we have the (now configurable!) delay for.

Reply:
For example, if something is wrong or very slow with the leader election, `JobManagerDeploymentStatus` is `READY ` but the session cluster is not ready for accepting job submission. What I mean is to update the `JobManagerDeploymentStatus` the `READY` only when the flink cluster is actually working.

Reply:
I agree, the current check is not very robust and works only in a very specific scenario. Let's open a ticket to track this effort and see if we can come up with something better :)

Reply:
I have created a ticket FLINK-26605 to track this effort.